### PR TITLE
CSP-29 Implements select a standard for add standard journey

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddAStandard/SelectAStandardController.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        [Route("{ukprn}/standards/new-standard", Name = RouteNames.PostAddStandardSelectStandard)]
+        [HttpPost]
+        public async Task<IActionResult> SubmitAStandard(SelectAStandardSubmitModel submitModel)
+        {
+            if (!ModelState.IsValid)
+            {
+                var model = await GetModel();
+                return View(ViewPath, model);
+            }
+            _logger.LogInformation("Begin of jounrey for ukprn: {ukprn} to add standard {larscode}", Ukprn, submitModel.SelectedLarsCode);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddAStandard/SelectAStandardController.cs:45]: typo: jounrey > journey

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/SelectAStandardControllerTests/SelectAStandardControllerGetTests.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            var response = await sut.SelectAStandard();
+
+            var viewResult = response as ViewResult;
+            Assert.IsNotNull(viewResult);
+            viewResult.ViewName.Should().Be(SelectAStandardController.ViewPath);
+            var model = viewResult.Model as SelectAStandardViewModel;
+            model.CancelLink.Should().Be(TestConstants.DefaultUrl);
+            var expectedNames = queryResult.AvailableCourses.Select(s => $"{s.Title} (Level {s.Level})");
+            model.Standards.All(s => expectedNames.Contains(s.Text)).Should().BeTrue();
+        }
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/SelectAStandardControllerTests/SelectAStandardControllerGetTests.cs:43]: Can you add a check for model.CancelLink?

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/AddAStandard/SelectAStandardSubmitModelValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using NUnit.Framework;
+using SFA.DAS.Roatp.CourseManagement.Web.Models.AddAStandard;
+using SFA.DAS.Roatp.CourseManagement.Web.Validators.AddAStandard;
+
+namespace SFA.DAS.Roatp.CourseManagement.Web.UnitTests.Validators.AddAStandard
+{
+    [TestFixture]
+    public class SelectAStandardSubmitModelValidatorTests
+    {
+        [TestCase(0, false)]
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/AddAStandard/SelectAStandardSubmitModelValidatorTests.cs:11]: not sure how the 0 is giving an isValid = false here. 0 is not empty...

**Review by MarkFCain — APPROVED** [review]
A few minor comments/observations mainly in tests, update or ignore comments as you see fit

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/AddAStandard/SelectAStandardSubmitModelValidatorTests.cs (hunk 3 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/AddAStandard/SelectAStandardSubmitModelValidatorTests.cs:11]: yes zero is considered as empty for non nullable numbers

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddAStandard/SelectAStandardController.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddAStandard/SelectAStandardController.cs:45]: Fixed

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/SelectAStandardControllerTests/SelectAStandardControllerGetTests.cs (hunk 2 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/SelectAStandardControllerTests/SelectAStandardControllerGetTests.cs:43]: Line#40?

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**